packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol

```diff
@@ -0,0 +1,979 @@
+// NounsDAOLogicV2 adds:
+// - `quorumParamsCheckpoints`, which store dynamic quorum parameters checkpoints
+// to be used when calculating the dynamic quorum.
+// - `_setDynamicQuorumParams(DynamicQuorumParams memory params)`, which allows the
+// DAO to update the dynamic quorum parameters' values.
+// - `getDynamicQuorumParamsAt(uint256 blockNumber_)`
+// - Individual setters of the DynamicQuorumParams members:
+//    - `_setMinQuorumVotesBPS(uint16 newMinQuorumVotesBPS)`
+//    - `_setMaxQuorumVotesBPS(uint16 newMaxQuorumVotesBPS)`
+//    - `_setQuorumVotesBPSOffset(uint16 newQuorumVotesBPSOffset)`
+//    - `_setQuorumLinearCoefficient(uint32 newQuorumLinearCoefficient)`
+//    - `_setQuorumQuadraticCoefficient(uint32 newQuorumQuadraticCoefficient)`
+// - `minQuorumVotes` and `maxQuorumVotes`, which returns the current min and
+// max quorum votes using the current Noun supply.
+// - New `Proposal` struct member:
+//    - `totalSupply` used in dynamic quorum calculation.
+//    - `creationBlock` used for retrieving checkpoints of votes and dynamic quorum params. This now
+// allows changing `votingDelay` without affecting the checkpoints lookup.
+// - `quorumVotes(uint256 proposalId)`, which calculates and returns the dynamic
+// quorum for a specific proposal.
+// - `proposals(uint256 proposalId)` instead of the implicit getter, to avoid stack-too-deep error
+//
+// NounsDAOLogicV2 removes:
+// - `quorumVotes()` has been replaced by `quorumVotes(uint256 proposalId)`.
+
+pragma solidity ^0.8.6;
+
+import './NounsDAOInterfaces.sol';
+
+contract NounsDAOLogicV2 is NounsDAOStorageV2, NounsDAOEventsV2 {
+    /// @notice The name of this contract
+    string public constant name = 'Nouns DAO';
+
+    /// @notice The minimum setable proposal threshold
+    uint256 public constant MIN_PROPOSAL_THRESHOLD_BPS = 1; // 1 basis point or 0.01%
+
+    /// @notice The maximum setable proposal threshold
+    uint256 public constant MAX_PROPOSAL_THRESHOLD_BPS = 1_000; // 1,000 basis points or 10%
+
+    /// @notice The minimum setable voting period
+    uint256 public constant MIN_VOTING_PERIOD = 5_760; // About 24 hours
+
+    /// @notice The max setable voting period
+    uint256 public constant MAX_VOTING_PERIOD = 80_640; // About 2 weeks
+
+    /// @notice The min setable voting delay
+    uint256 public constant MIN_VOTING_DELAY = 1;
+
+    /// @notice The max setable voting delay
+    uint256 public constant MAX_VOTING_DELAY = 40_320; // About 1 week
+
+    /// @notice The lower bound of minimum quorum votes basis points
+    uint256 public constant MIN_QUORUM_VOTES_BPS_LOWER_BOUND = 200; // 200 basis points or 2%
+
+    /// @notice The upper bound of minimum quorum votes basis points
+    uint256 public constant MIN_QUORUM_VOTES_BPS_UPPER_BOUND = 2_000; // 2,000 basis points or 20%
+
+    /// @notice The upper bound of maximum quorum votes basis points
+    uint256 public constant MAX_QUORUM_VOTES_BPS_UPPER_BOUND = 4_000; // 4,000 basis points or 40%
```

Comment:
Forgot to mention, we can probably bump this up to 60%. Still low enough to prevent governance from being fully locked by a bad update and high enough that we'd likely never want to exceed it.

Reply:
fixed https://github.com/nounsDAO/nouns-monorepo/commit/01d3c306bb5f47af7e5970cf0d79be55f9b0b54e